### PR TITLE
Update containerd and runc

### LINKED ISSUE #25880: groups for user not reported correctly via id or /usr/bin/groups any more?
**Output of `docker version`:**

```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 57
Server Version: 1.12.1
Storage Driver: btrfs
 Build Version: Btrfs v4.4
 Library Version: 101
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay null bridge host
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-34-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.417 GiB
Name: bdj13
ID: Q4IX:7U2J:NCLS:DV5X:7UVX:GI2S:LXLE:ES2F:RVIG:JJHS:JTPK:WBBH
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on physical host

**Steps to reproduce the issue:**
1. docker run -it --name grouptest ubuntu /bin/bash
2. useradd -m jenkins -G games,staff
3. in other terminal: docker exec -it -u jenkins grouptest /bin/bash -l
4. id (or /usr/bin/groups)

**Describe the results you received:**
```
$ id
uid=1000(jenkins) gid=1000(jenkins) groups=1000(jenkins)
```
```
$ /usr/bin/groups
jenkins
```

**Describe the results you expected:**
Expected jenkins to be reported as part of groups jenkins, staff and games
```
$ grep jenkins /etc/group 
staff:x:50:jenkins
games:x:60:jenkins
jenkins:x:1000:
```

**Additional information you deem important (e.g. issue happens only occasionally):**
Used to work with docker 1.10.x

### PR DESCRIPTION
**Pull request by mlaventure**
Signed-off-by: Kenfe-Mickael Laventure mickael.laventure@gmail.com
## 

Fixes #25880, #25470, #26995

### CODE REVIEW

**Comment by justincormack** [comment]
LGTM

**Comment by justincormack** [comment]
Doesn't update the libcontainer vendoring, do we want to do that too?

**Comment by mlaventure** [comment]
Oh, yes, thanks for pointing it out!

**Comment by vieux** [comment]
LGTM

**Comment by mlaventure** [comment]
There's another bug in `containerd-shim` that I'm in the process of fixing. If we could wait until I update this. The PR is [here](https://github.com/docker/containerd/pull/326)

**Comment by mlaventure** [comment]
Updated with aforementioned fix.

**Comment by LK4D4** [comment]
@mlaventure hah, rebase :)
LGTM though

**Comment by mlaventure** [comment]
Rebased :)

File: hack/vendor.sh [hunk 1]
```diff
@@ -106,7 +106,7 @@ clone git github.com/miekg/pkcs11 df8ae6ca730422dba20c768ff38ef7d79077a59f
 clone git github.com/docker/go v1.5.1-1-1-gbaf439e
 clone git github.com/agl/ed25519 d2b94fd789ea21d12fac1a4443dd3a3f79cda72c
 
-clone git github.com/opencontainers/runc cc29e3dded8e27ba8f65738f40d251c885030a28 # libcontainer
+clone git github.com/opencontainers/runc  02f8fa7863dd3f82909a73e2061897828460d52f # libcontainer
```
Comment (LK4D4) [inline hack/vendor.sh]: some leading space here

**Review by LK4D4 — CHANGES_REQUESTED** [review]

File: hack/vendor.sh (hunk 1 above)
Comment (LK4D4) [inline hack/vendor.sh]: Containerd commit should be changed here too I think

**Review by LK4D4 — APPROVED** [review]

File: hack/vendor.sh (hunk 1 above)
Comment (mlaventure) [inline hack/vendor.sh]: ditto

**Comment by justincormack** [comment]
The commit for github.com/godbus/dbus in https://github.com/opencontainers/runc/blob/master/Godeps/Godeps.json is v3, not v4.0.0 as it is on docker/docker which is probably an error - noticed this when looking at the vendoring before.

**Comment by justincormack** [comment]
Also github.com/syndtr/gocapability needs updating and also maybe github.com/golang/protobuf/proto

**Comment by mlaventure** [comment]
@justincormack 

- `libnetwork` is using v4 for `godbus/dbus` (and runc v3), hence the difference
- for `golang/protobuf`, `containerd` has the most recent vendoring followed by `swarmkit` then `runc` (`docker/docker` has the one from `containerd` atm)
- `libnetwork` is using a older commit than `runc` for `syndtr/gocapability` (which is the one used in `docker/docker` atm)

How do we reconcile this mess? :sweat_smile:

**Comment by LK4D4** [comment]
@mlaventure I'd better ignore deps of deps for now and use docker as source of truth - it has most extensive tests among all other projects.

**Comment by thaJeztah** [comment]
ping @mlaventure - per https://github.com/docker/docker/issues/26995#issuecomment-251742744 this fixes a regression in 1.12.2; are all these changes good to go into 1.12.2?

**Comment by justincormack** [comment]
@thaJeztah there are new features in here eg ambient capabilities.

**Comment by justincormack** [comment]
@mlaventure ugh. ok. LGTM, as I don't think this should matter now for the uses here.

**Comment by mlaventure** [comment]
@thaJeztah going to have to make a different PR for 1.12.2, the `containerd` used in it is based on a branch to avoid pulling all the GRPC changes.

I've opened a PR [here](https://github.com/docker/containerd/pull/327)

Will make one on 1.12.x branch once it's merged

**Comment by thaJeztah** [comment]
@mlaventure clear, thanks!

**Comment by thaJeztah** [comment]
ping @mlaventure - was a bump / Cherry-pick for 1.12.2 created?

**Comment by justincormack** [comment]
Yes I think so.

**Comment by mlaventure** [comment]
@thaJeztah 

Yes it was: https://github.com/docker/docker/pull/27196

**Pull request merged**